src/lib/statistics/cardinality_estimator.cpp

```diff
@@ -782,6 +785,90 @@ std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_inner_equi_join(
   return output_table_statistics;
 }
 
+std::shared_ptr<TableStatistics> CardinalityEstimator::estimate_semi_join(
+    const ColumnID left_column_id, const ColumnID right_column_id, const TableStatistics& left_input_table_statistics,
+    const TableStatistics& right_input_table_statistics) {
+  // This is based on estimate_inner_equi_join. We take the histogram from the right, set the bin heights to the
+  // distinct counts and run an inner/equi estimation on it. As there are no more duplicates on the right side, we
+  // should get the correct estimation for the left side.
+  const auto left_data_type = left_input_table_statistics.column_data_type(left_column_id);
+  const auto right_data_type = right_input_table_statistics.column_data_type(right_column_id);
+
+  // TODO(anybody) - Implement join estimation for differing column data types
+  //               - Implement join estimation for String columns
+  if (left_data_type != right_data_type || left_data_type == DataType::String) {
```

Comment:
Please help me to understand why the data type has an impact on the estimation.

Reply:
It doesn't, but we need to cast the statistics to a certain type.

Reply:
So that check and the following cross product estimation does not make sense, right?

Reply:
Good catch. We should forward the left side statistics.

Reply:
Can you add a comment why we forward them? I still don't get it. :)

Reply:
```
  // We expect both columns to be of the same type. This allows us to resolve the type only once, reducing the
  // compile time. For differing column types and/or string columns (which we cannot handle right now), we assume that
  // all tuples qualify. This is probably a gross overestimation, but we need to return something...

```

Reply:
Open an issue for the string estimation?